bazel, ui: bazel watch target for db-console

koorosh opened the pull request:
This change extends bazel build for UI to run
Db Console in watch mode. Currently, it has a
limitation to watch changes only for db-console
package and cannot track changes for cluster-ui
package (pkg/ui/workspaces/cluster-ui) due to
known `ibazel` issue (https://github.com/bazelbuild/rules_nodejs/issues/2107) with linked packages.

Depends on: https://github.com/cockroachdb/yarn-vendored/pull/84

Related: https://github.com/cockroachdb/cockroach/issues/73305

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/73787)
<!-- Reviewable:end -->

koorosh committed:
bazel, ui: bazel watch target for db-console

This change extends bazel build for UI to run
Db Console in watch mode. Currently, it has a
limitation to watch changes only for db-console
package and cannot track changes for cluster-ui
package (pkg/ui/workspaces/cluster-ui) due to
known ibazel issue with webpack linked packages.

Release note: None

koorosh commented on pkg/ui/workspaces/db-console/BUILD.bazel:
```diff
@@ -242,3 +242,21 @@ nodejs_binary(
+        "--env.dist=ccl",
+        "--env.is_bazel_build",
+        "--env.is_bazel_run",
+        "--env.target=http://localhost:8080",
```
`target` and `port` should be provided as args

rickystewart commented:
Is this change ready for review?

koorosh commented:
> Is this change ready for review?

Unfortunately, no. This change allows to build and watch for changes until command runs again (as described above).
Currently looking into more robust solution.
